soc/arm/nxp_imx/mimx8mm6_m4/CMakeLists.txt

```diff
@@ -7,3 +7,8 @@
 zephyr_sources(
   soc.c
   )
+
+if(CONFIG_OPENAMP_RSC_TABLE)
+  zephyr_linker_section(NAME .resource_table GROUP ROM_REGION NOINPUT)
```

Comment:
Based on the documentation for `zephyr_linker_section`, shouldn't the addition to the SOC linker script be unnecessary?

Reply:
I'm not really an expert, but at least what I can say is that if this entry is removed from the SoC linker script, the linker will complain with warnings and the table will be placed in a data section instead of a code section.

This [discussion](https://github.com/zephyrproject-rtos/zephyr/pull/42440#discussion_r826035005) may bring more details.

Reply:
That clears it up-wasn't aware that that this definition would only work if `DCONFIG_CMAKE_LINKER_GENERATOR` was set.